src/csrf.js

```diff
@@ -0,0 +1,75 @@
+import { defaults, isFunction, isString, result } from 'underscore';
+import Backbone from 'backbone';
+import Cookies from 'js-cookie';
+
+// URLs starting with // or http: are not relative to the same host.
+var nonRelative = /^(https?:)?\/\//i;
+// ... unless that is exactly what comes next.
+var sameOrigin = RegExp(nonRelative.source + window.location.host + '($|/)', 'i');
+
+// Expected types of the arguments to `wrapWithCSRF`.
+var requirements = {
+    sync: 'function with the same interface as Backbone.sync',
+    header: 'nonempty string',
+    cookie: 'nonempty string',
+};
+
+// Internal shorthand for the thing we do every time an argument has the wrong
+// type.
+function bail(key) {
```

Comment:
why not call the function `throwError` then instead?

Reply:
Purely out of habit. I saw somebody else use the word "bail" for this purpose and then started doing it myself. If you think that `throwError` is much clearer, I will change it.

Reply:
It's probably a matter of habit indeed. I haven't seen `bail` used anywhere yet.